cmd/definitions/generate.go

```diff
@@ -42,6 +43,63 @@ func generateGlobal(data *Data) {
 	generateT(objectT, "types/object.generated.go", data)
 }
 
+func generatePair(data *Data) *codegen.File {
+	f := codegen.
+		NewFile("pairs").
+		WithComment("Code generated by go generate cmd/definitions; DO NOT EDIT.").
+		AddImports(
+			codegen.NewImport("context"),
+			codegen.NewImport("time"),
+			codegen.NewImport("github.com/beyondstorage/go-storage/v4/pkg/httpclient"),
+			codegen.NewDotImport("github.com/beyondstorage/go-storage/v4/types"),
+		)
+
+	ps := make([]*Pair, 0, len(data.Pairs))
+	for _, v := range data.Pairs {
+		v := v
+		ps = append(ps, v)
+	}
+	sort.SliceStable(ps, func(i, j int) bool {
+		return ps[i].Name < ps[j].Name
+	})
+
+	for _, v := range ps {
+		pname := templateutils.ToPascal(v.Name)
+
+		fn := codegen.NewFunction("With" + pname)
+
+		fn.WithComment(fmt.Sprintf(`
+With%s will apply %s value to Options.
+
+%s %s
+`, pname, v.Name, pname, v.originalDescription))
+
+		fn.WithParameters(codegen.NewParameter("v", v.Type(), false))
+		fn.WithResults(codegen.NewResult("p", "Pair", false))
+		fn.WithBody(`
+return Pair{
+	Key: "{{ .Name }}",
+	Value: v,
+}
+`, v)
+
+		f.AddFunction(fn)
```

Comment:
It's more readable and easy to debug. But I wondered whether the whole source file will be complicated for more  complex function body and more complex types?

Reply:
What do you mean by "easy to debug"?

Reply:
> more complex function body and more complex types

It's indeed a question, I can't answer for now (:

Reply:
I have encountered errors during code generation when using templates before and could not visually locate the error in the template. I think maybe this approach will be easier in dealing with this type of problem？